src/Microsoft.ML.CpuMath/SseIntrinsics.cs

```diff
@@ -102,6 +444,72 @@ internal static unsafe void ScaleU(float scale, Span<float> dst)
             }
         }
 
+        internal static unsafe void ScaleSrcU(float scale, Span<float> src, Span<float> dst)
+        {
+            Vector128<float> scaleVector = Sse.SetAllVector128(scale);
+
+            fixed (float* psrc = src)
+            fixed (float* pdst = dst)
+            {
+                float* pDstEnd = pdst + dst.Length;
+                float* pSrcCurrent = psrc;
```

Comment:
We shouldn't need `pSrcCurrent` and `pDstCurrent`, right?  We can just use `psrc` and `pdst` here.

Reply:
This time, it may be different from the case in the matrix-op function (e.g. `MatMulA`).  Since here `psrc` is fixed to point to `src` in the `fixed` statement, `psrc` cannot be modified or incremented inside the `fixed` block.  I assigned a new variable `pSrcCurrent` the value of `psrc`, so that we can increment `pSrcCurrent` to achieve the desired behavior.